### PR TITLE
OSDOCS-12664: Adding worker node instance types.

### PR DESCRIPTION
**Pull request by eohartman**
Version(s):
4.18

Issue:
[OSDOCS-12664](https://issues.redhat.com/browse/OSDOCS-12664)

Link to docs preview:

- [ROSA AWS instance types](https://85259--ocpdocs-pr.netlify.app/openshift-rosa-hcp/latest/rosa_architecture/rosa_policy_service_definition/rosa-hcp-instance-types.html#rosa-sdpolicy-aws-instance-types_rosa-hcp-instance-types)

- [ROSA AWS Arm-based Graviton instance types](https://85259--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_architecture/rosa_policy_service_definition/rosa-hcp-instance-types.html#rosa-sdpolicy-aws-instance-types-graviton_rosa-hcp-instance-types)

- [OSD AWS instance types for Customer Cloud Subscription clusters](https://85259--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_architecture/osd_policy/osd-service-definition#aws-instance-types-ccs_osd-service-definition)

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by eohartman** [comment]
@xueli181114 Can I please get a QE review of this PR? Thank you!

**Comment by arendej** [comment]
the x8g and r8g series belong in the Graviton category.

**Review by arendej — CHANGES_REQUESTED** [review]
the r8g and x8g series belong in the graviton category

**Comment by AkashKanni** [comment]
@eohartman I will work on QE review for this. Thank you! cc : @xueli181114

**Comment by eohartman** [comment]
@arendej Can you please take a look at the previews above and verify that the information is in the correct locations? Thank you!

**Review by arendej — APPROVED** [review]
lgtm

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc [hunk 1]
```diff
@@ -66,6 +66,16 @@ Graviton instance types are only available for new clusters created after 24 Jul
...
 - x2gd.metal (64 vCPU, 1024 GiB)
+- x8g.xlarge (4 vCPU, 64 GiB)
+- x8g.2xlarge (8 vCPU, 128 GiB)
+- x8g.4xlarge (16 vCPU, 256 GiB)
+- x8g.8xlarge (32 vCPU, 512 GiB)
+- x8g.12xlarge (48 vCPU, 768 GiB)
+- x8g.16xlarge (64 vCPU, 1,024 GiB)
+- x8g.24xlarge (96 vCPU, 1,536 GiB)
+- x8g.48xlarge (192 vCPU, 3,072 GiB)
+- x8g.metal-24xl	 (96 vCPU, 1,536 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:77]: This renders correctly, but I'm not sure the code is correct with the large space between the instance type and information in parentheses.

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc [hunk 2]
```diff
@@ -66,6 +66,16 @@ Graviton instance types are only available for new clusters created after 24 Jul
...
+- x8g.metal-48xl	 (192 vCPU, 3,072 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:78]: See above comment.

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc [hunk 3]
```diff
@@ -99,6 +109,16 @@ Graviton instance types are only available for new clusters created after 24 Jul
...
 - r7gd.metal (64 vCPU, 512 GiB)
+- r8g.xlarge (4 vCPU, 32 GiB)
+- r8g.2xlarge (8 vCPU, 64 GiB)
+- r8g.4xlarge (16 vCPU, 128 GiB)
+- r8g.8xlarge (32 vCPU, 256 GiB)
+- r8g.12xlarge (48 vCPU, 384 GiB)
+- r8g.16xlarge (64 vCPU, 512 GiB)
+- r8g.24xlarge (96 vCPU, 768 GiB)
+- r8g.48xlarge (192 vCPU, 1,536 GiB)
+- r8g.metal-24xl	(96 vCPU, 768 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:120]: Unsure about spacing, see above comments.

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc [hunk 4]
```diff
@@ -99,6 +109,16 @@ Graviton instance types are only available for new clusters created after 24 Jul
...
+- r8g.metal-48xl	(192 vCPU, 1,536 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:121]: Unsure about spacing, see above comments.

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc [hunk 5]
```diff
@@ -66,6 +66,16 @@ Graviton instance types are only available for new clusters created after 24 Jul
 - x2gd.12xlarge (32 vCPU, 768 GiB)
 - x2gd.16xlarge (64 vCPU, 1024 GiB)
 - x2gd.metal (64 vCPU, 1024 GiB)
+- x8g.xlarge (4 vCPU, 64 GiB)
+- x8g.2xlarge (8 vCPU, 128 GiB)
+- x8g.4xlarge (16 vCPU, 256 GiB)
+- x8g.8xlarge (32 vCPU, 512 GiB)
+- x8g.12xlarge (48 vCPU, 768 GiB)
+- x8g.16xlarge (64 vCPU, 1,024 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:74]:
```suggestion
- x8g.16xlarge (64 vCPU, 1024 GiB)
```
Instance types don't have commas in the numbers.

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc [hunk 6]
```diff
@@ -66,6 +66,16 @@ Graviton instance types are only available for new clusters created after 24 Jul
...
+- x8g.24xlarge (96 vCPU, 1,536 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:75]:
```suggestion
- x8g.24xlarge (96 vCPU, 1536 GiB)
```

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc [hunk 7]
```diff
@@ -66,6 +66,16 @@ Graviton instance types are only available for new clusters created after 24 Jul
...
+- x8g.48xlarge (192 vCPU, 3,072 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:76]:
```suggestion
- x8g.48xlarge (192 vCPU, 3072 GiB)
```

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc (hunk 1 above)
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:77]:
```suggestion
- x8g.metal-24xl	 (96 vCPU, 1536 GiB)
```

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc (hunk 2 above)
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:78]:
```suggestion
- x8g.metal-48xl	 (192 vCPU, 3072 GiB)
```

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc [hunk 8]
```diff
@@ -99,6 +109,16 @@ Graviton instance types are only available for new clusters created after 24 Jul
...
 - r7gd.16xlarge (64 vCPU, 512 GiB)
 - r7gd.metal (64 vCPU, 512 GiB)
+- r8g.xlarge (4 vCPU, 32 GiB)
+- r8g.2xlarge (8 vCPU, 64 GiB)
+- r8g.4xlarge (16 vCPU, 128 GiB)
+- r8g.8xlarge (32 vCPU, 256 GiB)
+- r8g.12xlarge (48 vCPU, 384 GiB)
+- r8g.16xlarge (64 vCPU, 512 GiB)
+- r8g.24xlarge (96 vCPU, 768 GiB)
+- r8g.48xlarge (192 vCPU, 1,536 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:119]:
```suggestion
- r8g.48xlarge (192 vCPU, 1536 GiB)
```

File: modules/rosa-sdpolicy-am-aws-compute-types.adoc [hunk 9]
```diff
@@ -315,6 +314,25 @@
...
+- g6.24xlarge (96 vCPU, 384 GiB)
+- g6.48xlarge (192 vCPU, 768 GiB)
+- g6e.xlarge (4 vCPU, 32 GiB)
+- g6e.2xlarge (8 vCPU, 64 GiB)
+- g6e.4xlarge (16 vCPU, 128 GiB)
+- g6e.8xlarge (32 vCPU, 256 GiB)
+- g6e.16xlarge (64 vCPU, 512 GiB)
+- g6e.12xlarge (48 vCPU, 384 GiB)
+- g6e.24xlarge (96 vCPU, 768 GiB)
+- g6e.48xlarge (192 vCPU, 1,536 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types.adoc:334]:
```suggestion
- g6e.48xlarge (192 vCPU, 1536 GiB)
```

File: modules/rosa-sdpolicy-am-aws-compute-types.adoc [hunk 10]
```diff
@@ -315,6 +314,25 @@
 - g5.24xlarge (96 vCPU, 384 GiB)
 - g5.48xlarge (192 vCPU, 768 GiB)
 - dl1.24xlarge  (96 vCPU, 768 GiB)&#8224;
+- g6.xlarge (4 vCPU, 16 GiB)
+- g6.2xlarge (8 vCPU, 32 GiB)
+- g6.4xlarge (16 vCPU,	64 GiB)
+- g6.8xlarge (32 vCPU, 128 GiB)
+- g6.16xlarge (64 vCPU, 256 GiB)
+- gr6.4xlarge (16 vCPU, 128 GiB)
+- gr6.8xlarge (32 vCPU, 256 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types.adoc:323]: Is there a reason the two `gr6` instance types (currently lines 322 and 323) are mixed in with the `g6`s. I think all the `g6` types should be listed in order, then the `gr6`s and then the `g6e` to make it easier to read.

File: modules/rosa-sdpolicy-am-aws-compute-types.adoc [hunk 11]
```diff
@@ -315,6 +314,25 @@
 - g5.24xlarge (96 vCPU, 384 GiB)
 - g5.48xlarge (192 vCPU, 768 GiB)
 - dl1.24xlarge  (96 vCPU, 768 GiB)&#8224;
+- g6.xlarge (4 vCPU, 16 GiB)
+- g6.2xlarge (8 vCPU, 32 GiB)
+- g6.4xlarge (16 vCPU,	64 GiB)
+- g6.8xlarge (32 vCPU, 128 GiB)
+- g6.16xlarge (64 vCPU, 256 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types.adoc:321]: g6.16xlarge should go after g6.12xlarge.

File: modules/rosa-sdpolicy-am-aws-compute-types.adoc [hunk 12]
```diff
@@ -315,6 +314,25 @@
...
+- gr6.4xlarge (16 vCPU, 128 GiB)
+- gr6.8xlarge (32 vCPU, 256 GiB)
+- g6.12xlarge (48 vCPU, 192 GiB)
+- g6.24xlarge (96 vCPU, 384 GiB)
+- g6.48xlarge (192 vCPU, 768 GiB)
+- g6e.xlarge (4 vCPU, 32 GiB)
+- g6e.2xlarge (8 vCPU, 64 GiB)
+- g6e.4xlarge (16 vCPU, 128 GiB)
+- g6e.8xlarge (32 vCPU, 256 GiB)
+- g6e.16xlarge (64 vCPU, 512 GiB)
```
Comment (jneczypor) [inline modules/rosa-sdpolicy-am-aws-compute-types.adoc:331]: 16xlarge should go after 12xlarge.

File: modules/sdpolicy-am-aws-compute-types-ccs.adoc [hunk 13]
```diff
@@ -305,7 +305,24 @@
 - g5.24xlarge (96 vCPU, 384 GiB)
 - g5.48xlarge (192 vCPU, 768 GiB)
 - dl1.24xlarge  (96 vCPU, 768 GiB)&#8224;
-
+- g6.xlarge (4 vCPU, 16 GiB)
+- g6.2xlarge (8 vCPU, 32 GiB)
+- g6.4xlarge (16 vCPU,	64 GiB)
+- g6.8xlarge (32 vCPU, 128 GiB)
+- g6.16xlarge (64 vCPU, 256 GiB)
+- gr6.4xlarge (16 vCPU, 128 GiB)
```
Comment (jneczypor) [inline modules/sdpolicy-am-aws-compute-types-ccs.adoc:313]: Please pull the `gr6`s out. See above comment.

File: modules/sdpolicy-am-aws-compute-types-ccs.adoc [hunk 14]
```diff
@@ -305,7 +305,24 @@
...
+- gr6.8xlarge (32 vCPU, 256 GiB)
+- g6.12xlarge (48 vCPU, 192 GiB)
```
Comment (jneczypor) [inline modules/sdpolicy-am-aws-compute-types-ccs.adoc:315]: Put 12xlarge before 16xlarge. See above comment.

File: modules/sdpolicy-am-aws-compute-types-ccs.adoc [hunk 15]
```diff
@@ -305,7 +305,24 @@
...
+- g6.24xlarge (96 vCPU, 384 GiB)
+- g6.48xlarge (192 vCPU, 768 GiB)
+- g6e.xlarge (4 vCPU, 32 GiB)
+- g6e.2xlarge (8 vCPU, 64 GiB)
+- g6e.4xlarge (16 vCPU, 128 GiB)
+- g6e.8xlarge (32 vCPU, 256 GiB)
+- g6e.16xlarge (64 vCPU, 512 GiB)
```
Comment (jneczypor) [inline modules/sdpolicy-am-aws-compute-types-ccs.adoc:322]: Put 16xlarge after 12xlarge. See above comments.

File: modules/sdpolicy-am-aws-compute-types-ccs.adoc [hunk 16]
```diff
@@ -305,7 +305,24 @@
...
+- g6e.12xlarge (48 vCPU, 384 GiB)
+- g6e.24xlarge (96 vCPU, 768 GiB)
+- g6e.48xlarge (192 vCPU, 1,536 GiB)
```
Comment (jneczypor) [inline modules/sdpolicy-am-aws-compute-types-ccs.adoc:325]:
```suggestion
- g6e.48xlarge (192 vCPU, 1536 GiB)
```

**Review by jneczypor — COMMENTED** [review]
A few adjustments to change before merging. Thanks!

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

File: modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc (hunk 5 above)
Comment (eohartman) [reply modules/rosa-sdpolicy-am-aws-compute-types-graviton.adoc:74]: They have commas in the numbers on AWS site: https://aws.amazon.com/ec2/instance-types/

**Comment by eohartman** [comment]
@jneczypor Can you please take a look at the changes to make sure I caught everything you mentioned? Thank you!

**Comment by jneczypor** [comment]
Hi @eohartman - My peer review week finished last Friday. I'll re-tag the ticket so that someone who is on peer review this week can take a look. Or you can just move it to merge review. Thanks!

**Review by skrthomas — COMMENTED** [review]
Out of scope comment, but on line 8, I notice just x86 is used. In the SSG, it says this is an incorrect form and to use x86_64 instead: https://redhat-documentation.github.io/supplementary-style-guide/#x86_64.

Otherwise the changes lgtm!

**Comment by eohartman** [comment]
@skrthomas Thank you! My guess is that this is a problem in other places as well, so I'm going to create a separate ticket for it so I can make sure and catch all the instances (pun intended) of it. :-)

**Pull request merged**